Base/Python/slicer/parameterNodeWrapper.py

```diff
@@ -0,0 +1,814 @@
+        Note that most implementations can just use parameterNode.HasParameter(name), but certain ones
+        like NodeSerializer cannot.
+        """
+        raise NotImplementedError("isIn is not implemented")
+
+    @abc.abstractmethod
+    def write(self, parameterNode, name: str, value) -> None:
+        """
+        Writes the value to the parameterNode under the given name.
+        Note: It is acceptable to mangle the name as long the same name can be used for reading.
+        For example ListSerializer does this.
+        """
+        raise NotImplementedError("write is not implemented")
+
+    @abc.abstractmethod
+    def read(self, parameterNode, name: str):
+        """
+        Reads and returns the value with the given name from the parameterNode.
+        """
+        raise NotImplementedError("read is not implemented")
+
+    @abc.abstractmethod
+    def remove(self, parameterNode, name: str) -> None:
+        """
+        Removes the value of the given name from the parameterNode.
+        """
+        raise NotImplementedError("remove is not implemented")
+
+    @staticmethod
+    def supportsCaching():
+        """
+        Whether this serializer works with caching.
+
+        Converting from str to a value can be costly, so caching can be used to speed up reads.
+        However, if a cached value is returned and that value is an class object, modifying the
+        returned object can modify the cache without modifying the underlying parameterNode,
+        which can lead to unexpected behavior.
+
+        Serializers that support caching typically return objects that are able to update the
+        parameterNode (usually through this serializer) whenever their state changes. See
+        ListSerializer and ObservedList for an example. Alternatively, types that do not return by
+        reference (int, float) or are immutable (str) are good candidates for caching.
+        """
+        return False
+
+
+class NumberSerializer(Serializer):
+    """
+    Serializer for numeric types (e.g. int, float).
+    """
+    def __init__(self, type):
+        """
+        Constructs a serializer for the given numeric type.
+        """
+        self.type = type
+
+    def isIn(self, parameterNode, name: str) -> bool:
+        return parameterNode.HasParameter(name)
+
+    def write(self, parameterNode, name: str, value) -> None:
+        parameterNode.SetParameter(name, str(value))
+
+    def read(self, parameterNode, name: str):
+        return self.type(parameterNode.GetParameter(name))
+
+    def remove(self, parameterNode, name: str) -> None:
+        parameterNode.UnsetParameter(name)
+
+    @staticmethod
+    def supportsCaching():
+        return True
+
+
+class StringSerializer(Serializer):
+    """
+    Serializer for str. This is the type that parameterNode.GetParameter/SetParameter use
+    so it is mainly a pass through.
+    """
+    @staticmethod
```

Comment:
Nitpick - Making a method static in a derived class seems like a code smell to me; the signature isn't the same anymore. It seems to work, though, so it might not be an issue.

Reply:
I can update to not be static to match.

Reply:
See https://github.com/Slicer/Slicer/pull/6684/commits/0de798353a335df4f47c58b36e8d3eec58520dfc